src/hotspot/share/opto/compile.hpp

```diff
@@ -829,18 +829,40 @@ class Compile : public Phase {
 
   const CompilationFailureInfo* first_failure_details() const { return _first_failure_details; }
 
+  bool failing(bool skip=false) {
+    if (failing_internal()) {
+      return true;
+    }
+    if (!StressBailout || skip) {
+      return false;
+    }
+    return fail_randomly(StressBailoutInterval);
+  }
+
+  bool fail_randomly(uint invprob) {
+    guarantee(0 < invprob, "domain error");
+#ifdef ASSERT
+    return false; // debug builds assert on bailouts.
+#endif
+    if (!_stress_seed || (random() % invprob)) {
```

Comment:
It was intended to be a workaround until JDK-8333956 can be pushed, to check if the stress seed has been initialized yet -- but it's a good point that StressSeed=0 is now excluded. We could use an additional variable to indicate that the stress seed has been initialized.

Reply:
Yes, or just declare StressSeed=0 being invalid. I dimly remember running into problems with os::random and 0 seeds at some time. Cannot remember the issue, but since then, I try to avoid 0 seeds.